run_unittests.py

```diff
@@ -924,11 +924,11 @@ def setUp(self):
         # Misc stuff
         self.orig_env = os.environ.copy()
         if self.backend is Backend.ninja:
-            self.no_rebuild_stdout = 'ninja: no work to do.'
+            self.no_rebuild_stdout = ['ninja: no work to do.', 'samu: nothing to do']
         else:
             # VS doesn't have a stable output when no changes are done
             # XCode backend is untested with unit tests, help welcome!
-            self.no_rebuild_stdout = 'UNKNOWN BACKEND {!r}'.format(self.backend.name)
+            self.no_rebuild_stdout = ['UNKNOWN BACKEND {!r}'.format(self.backend.name)]
```

Comment:
I wasn't sure what to do here, since `self.no_rebuild_stdout` is only inspected if the backend is ninja. Maybe it should just set it to `None`, and then change line 1136 to `if self.no_rebuild_stdout:`?